### ISSUE TITLE
don't treat objects without stroke as SatinColumn

### ISSUE THREAD

**Issue by fnavas00** [inkstitch/inkstitch#460]
I keep seeing this error message I have no idea what it means im having an extremely difficult time learning to use this extension I keep getting error message after error message none of the embroidex colors match the ones I have. There is no white and light blue all the thread colors are off. I need help. Why do I keep seeing this error message "path5049: error: satin column: object path5049 has a fill (but should not)" There is no way to turn off the fill ive tried and it erases the thread color of my svg so im only left with the outline of it. Then I get error message 

"Traceback (most recent call last):
  File "inkstitch.py", line 54, in <module>
    stderr = stderr.strip()
  File "inkscape/share/extensions/inkex.py", line 283, in affect
  File "lib/extensions/simulate.py", line 17, in effect
  File "lib/extensions/base.py", line 181, in elements_to_patches
  File "lib/elements/element.py", line 268, in embroider
  File "lib/elements/satin_column.py", line 772, in to_patches
  File "lib/elements/satin_column.py", line 328, in validate_satin_column
IndexError: list index out of range"

 what do i do after that how do I get thread color on my art if I keep getting the path5049: error: satin column: object path5049 has a fill (but should not)

**Comment by kaalleen** [comment]
Welcome to the project @fnavas00

Satin columns get their color definition from the stroke color (not from the fill color). So you are on the right way, if you are seeing only the outline. The color of the "outline" will be used as the satin thread color.

For the error message you are receiving, have a look at [an other issue](https://github.com/inkstitch/inkstitch/issues/459#issuecomment-491461441), where you can find this exact error message with a solution.

We have more information about stitch types and how to create them on our website: https://inkstitch.org/docs/stitches/satin-column/

If you rather watch a video, have a look at our video tutorial about satin columns: https://www.youtube.com/watch?v=U7YtajKlftQ&list=PLvlbfDmZyXG1ORmeqHdp4aP7J71e7icJP&index=7

**Comment by fnavas00** [comment]
Im not trying to use satin column though Im trying to do a fill for this shape

**Comment by jameskolme** [comment]
Familiar error to me, but actually I can´t repeat it when trying to do so. Every time I get this same error tinkering with enable/disable fill or stroke fixes it. Not sure what is causing this and this is not that common error.

**Comment by fnavas00** [comment]
> Familiar error to me, but actually I can´t repeat it when trying to do so. Every time I get this same error tinkering with enable/disable fill or stroke fixes it. Not sure what is causing this and this is not that common error.

Doesnt work

**Comment by jameskolme** [comment]
Can you upload small sample from .svg that is still causing this error?

**Comment by fnavas00** [comment]
> Can you upload small sample from .svg that is still causing this error?

Here's a link to the file https://drive.google.com/open?id=1VdmbPtyrg6hcgtOxVx0Ovdtnmeqg-X5m

**Comment by jameskolme** [comment]
path5045: error: satin column: object path5045 has a fill (but should not) I get this error, that is great. 

Possible solution:
select all 
go to object/fill and stroke, stroke paint and select no paint 
now go to inkstitch/params (fill should be recommend and simulate takes some time to start)
apply and quit
This works for me, 4-5 smallest areas are actually causing this, not sure why that is.

**Comment by fnavas00** [comment]
> path5045: error: satin column: object path5045 has a fill (but should not) I get this error, that is great.
> 
> Possible solution:
> select all
> go to object/fill and stroke, stroke paint and select no paint
> now go to inkstitch/params (fill should be recommend and simulate takes some time to start)
> apply and quit
> This works for me, 4-5 smallest areas are actually causing this, not sure why that is.

Didn't work, got this error message "path5045: error: satin column: object path5045 has a fill (but should not)"

**Comment by jameskolme** [comment]
Ok there is now something going, I´m trying to figure it out. It did work just like I described, more than once, but now I get satin column error hmm

**Comment by fnavas00** [comment]
It only works when I do individual pieces of the svg but not when I select all

**Comment by jameskolme** [comment]
Is this svg file imported from other program? It would explain why this is happening and why my latest solution seems to work (again).

Open **object/objects** and select objects exactly as shown, select all will not work, must be selected using mouse. 
![select](https://user-images.githubusercontent.com/44113605/57772249-a0f9e780-771d-11e9-9da9-4d466d1b81da.png)
**path/break apart** (something interesting should happen on object window)
**select all**
**simulate** should work

But did it?

Selecting paths and doing break apart resets path data somehow, this is not common procedure to make.

**Comment by fnavas00** [comment]
Yes it's a path imported from GIMP.. It worked!! thank you sooo much I was starting to get worried

**Comment by lexelby** [comment]
There's a lot going on here!  This is at least two bugs in Ink/Stitch and also a somewhat confusing user experience.

First bug: #366 is what's causing the "index out of range" error.  That bug is hitting a lot of folks and I plan to prioritize fixing it.

Next is the confusing user experience.  If you have a shape with both a stroke _and_ a fill set, Ink/Stitch will try to do two different things.  First, it will do fill stitch, and second, it will do zig-zag around the border.  

However, if you enable Satin Column instead, it will try to do a satin column.  But Satin Columns don't make sense with a fill, and Ink/Stitch complains at you "object has a fill but it should not".  That error message should be made much more clear.

That brings us to the second bug.  If Ink/Stitch sees the "satin column enabled" attribute on an object in the SVG, then it treats it as a satin column no matter what.  This happens even if you changed your mind and you no longer want the object to be a satin column!  For your SVG, if you disable the stroke on the objects, everything _should_ be fine.  But because of this bug, it still gives you the "object has a fill but it should not" error.

To work around this bug, you need to remove the "embroider_satin_column" XML attribute in the SVG.  @jameskolme, your "break apart" procedure fixed the problem because Inkscape removes the `embroider_` XML attributes when it breaks shapes apart, since those aren't standard SVG attributes.

Bottom line is, there are two thins to do for this ticket:

* rephrase the "object has a fill but it should not" message
* Ink/Stitch should ignore the `embroider_satin_column` attribute for shapes without a stroke.
